### PR TITLE
broaden gson range, bundle new gson and guava

### PR DESCRIPTION
**Pull request by cdietrich**
@jonahgraham do we need a dot release for simrel?

### CODE REVIEW

**Review by szarnekow — APPROVED** [review]

**Comment by szarnekow** [comment]
As a client of lsp4j I'd even consider using a 3.0 as the exclusive upper bound of the dependency.

**Comment by jonahgraham** [comment]
> As a client of lsp4j I'd even consider using a 3.0 as the exclusive upper bound of the dependency.

We got (minorly) burned a while ago because a new version of gson didn't strictly follow semantic versioning. Ever since then I have been concerned to not have a runtime problem.

However I am asked multiple times for this, so seeing as many consumers seem ok with the (minor) risk, I say lets widen it up.

> @jonahgraham do we need a dot release for simrel?

Yes

File: gradle/versions.gradle [hunk 1]
```diff
@@ -15,7 +15,7 @@ version = '0.24.0-SNAPSHOT'
 ext.versions = [
 	'xtend_lib': '2.32.0',
 	'guava': '[32.1.2,34)',
-	'gson': '[2.9.1,2.11)',
+	'gson': '[2.9.1,2.12)',
```
Comment (jonahgraham) [inline gradle/versions.gradle:18]:
```suggestion
	'gson': '[2.9.1,3.0)',

```

**Comment by cdietrich** [comment]
Can you merge am afk

**Pull request merged**